drivers/include/digit7seg.h

```diff
@@ -0,0 +1,202 @@
+    gpio_t data_g;  /**< Data input pin */
+    gpio_t data_dp; /**< Data input pin */
+    gpio_t dig1;    /**< Data input pin */
+    gpio_t dig2;    /**< Data input pin */
+    gpio_t dig3;    /**< Data input pin */
+    gpio_t dig4;    /**< Data input pin */
+    tim_t timer;    /**< Timer to use in digit7seg_power* */
+    uint8_t digits; /**< Number of digits */
+} digit7seg_params_t;
+
+/**
+ * @brief   Device descriptor for the DIGIT7SEG
+ */
+typedef struct {
+    digit7seg_params_t params; /**< Device parameters */
+    uint32_t value;            /**< Binary value to display */
+    uint8_t current_digit;     /**< Displayed digit */
+} digit7seg_t;
+
+/**
+ * @brief   Initialize the given DIGIT7SEG
+ *
+ * @param[out] dev      DIGIT7SEG device descriptor to initialize
+ * @param[in]  params   Device parameters to use
+ *
+ * @return  0 on success or negative error code, see #digit7seg_error_codes
+ */
+int digit7seg_init(digit7seg_t *dev, const digit7seg_params_t *params);
+
+/**
+ * @brief   Set the value for every digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value as an uint32_t
+ *
+ */
+void digit7seg_set_all_value(digit7seg_t *dev, uint32_t value);
+
+/**
+ * @brief   Set the value for one digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value as an uint8_t, a is equal to the first bit
+ *                       b the second.... dp the 8th bit.
+ *
+ * @return              0 on success
+ * @return              -1 when an invalid index is passed
+ */
+int digit7seg_set_value(digit7seg_t *dev, int index, uint8_t value);
+
+#if MODULE_DIGIT7SEG_INT || DOXYGEN
+/**
+ * @brief   Set the value for every digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value between [0, 9999]
+ *
+ */
+int digit7seg_set_int_all_value(digit7seg_t *dev, uint16_t value);
+
+/**
+ * @brief   Set the value for one digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value between [0, 9]
+ *
+ */
+int digit7seg_set_int_value(digit7seg_t *dev, int index, uint8_t value);
+#endif
+
+#if MODULE_DIGIT7SEG_FLOAT || DOXYGEN
+/**
+ * @brief   Set a float value dispatched between every display
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     A float value between [0, 999,9]
+ * @param[in] precision The multiplier factor ex: value = 234,15, precision = 10
+ *                       will be displayed as 234,1 but with
+ *                       precision = 100 it will be 34,15
+ */
+int digit7seg_set_float_value(digit7seg_t *dev, float value, int precision);
```

Comment:
We support a lot of MCUs that do not have an FPU. And soft FPU implementations are surprisingly large (in terms of ROM size) and also slowish.

How about providing the value as integer type and a second integer gives the position of the decimal?

E.g. `value=1337` and `decimal=2` would be printed as `13.37`?